pkg/ebpf/perf/event.go

```diff
@@ -0,0 +1,213 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux_bpf
+
+// Package perf implements types related to eBPF and the perf subsystem, like perf buffers and ring buffers.
+package perf
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/features"
+	"github.com/cilium/ebpf/perf"
+	"github.com/cilium/ebpf/ringbuf"
+
+	ebpfTelemetry "github.com/DataDog/datadog-agent/pkg/ebpf/telemetry"
+	ddsync "github.com/DataDog/datadog-agent/pkg/util/sync"
+)
+
+var perfPool = ddsync.NewDefaultTypedPool[perf.Record]()
+var ringbufPool = ddsync.NewDefaultTypedPool[ringbuf.Record]()
+
+// Flushable is an interface for objects that support flushing
+type Flushable interface {
+	Flush()
```

Comment:
nit: there are a lot of impls for this base method, maybe it can be named something like `FlushRecord` or something so it's easier to search for impls in an IDE

Reply:
That would require changing `cilium/ebpf`.